go/vt/vtctl/reparentutil/emergency_reparenter.go

```diff
@@ -518,13 +517,26 @@ func (erp *EmergencyReparenter) reparentReplicas(
 	rec := concurrency.AllErrorRecorder{}
 
 	handlePrimary := func(alias string, tablet *topodatapb.Tablet) error {
-		position, err := erp.tmc.PrimaryPosition(replCtx, tablet)
-		if err != nil {
-			return err
-		}
-		if populateReparentJournal {
+		if !intermediateReparent {
+			var position string
+			var err error
+			if ev.ShardInfo.PrimaryAlias == nil {
+				erp.logger.Infof("setting up %v as new primary for an uninitialized cluster", alias)
+				// we call InitPrimary when the PrimaryAlias in the ShardInfo is empty. This happens when we have an uninitialized cluster.
+				position, err = erp.tmc.InitPrimary(replCtx, tablet, SemiSyncAckers(opts.durability, tablet) > 0)
+			} else {
+				erp.logger.Infof("starting promotion for the new primary - %v", alias)
+				// we call PromoteReplica which changes the tablet type, fixes the semi-sync, set the primary to read-write and flushes the binlogs
+				position, err = erp.tmc.PromoteReplica(replCtx, tablet, SemiSyncAckers(opts.durability, tablet) > 0)
```

Comment:
typically we use RemoteOperationTimeout for RPCs. waitReplicasTimeout is special, it's passed in from the caller. It may be worth creating a separate context to call PromoteReplica with using RemoteOperationTimeout.

Reply:
Fixed 👍